lib/whitehall_importer.rb

```diff
@@ -23,9 +25,26 @@ def self.create_migration(organisation_content_id, document_type)
     end
   end
 
-  def self.import_and_sync(whitehall_document)
-    whitehall_import = WhitehallMigration::DocumentImport.create!(
-      whitehall_document_id: whitehall_document["id"],
+  def self.queue_migration_jobs(whitehall_migration)
+    documents = WhitehallMigration::DocumentImport.where(whitehall_migration_id: whitehall_migration.id)
+    documents.each do |document|
+      WhitehallDocumentImportJob.perform_later(document)
+    end
+  end
+
+  def self.mark_migration_completed(whitehall_migration_document_import)
+    running_states = %w(pending importing imported syncing)
+    whitehall_migration = WhitehallMigration.find(whitehall_migration_document_import.whitehall_migration_id)
+    if WhitehallMigration::DocumentImport.where(whitehall_migration_id: whitehall_migration.id, state: running_states).empty?
+      whitehall_migration.update!(
+        end_time: Time.current,
+      )
+    end
+  end
+
+  def self.import_and_sync(whitehall_import)
```

Comment:
Something to consider is that if this method is being called from ActiveJob then we should assume that we don't know what state the whitehall_import model is in (Jobs should be idempotent and resilient to retrying). 

We hit a problem here that this method immediately changes the state of the `whitehall_import` model which hinders the idempotency of the job. 

Might be that all we want to do is assert that the whitehall_import is pending when this method is called and worry further about retryability in the follow up story.